framework/PostProcessors/ParetoFrontierPostProcessor.py

```diff
@@ -0,0 +1,147 @@
+    """
+    inputSpecification = super(ParetoFrontier, cls).getInputSpecification()
+    inputSpecification.addSub(InputData.parameterInputFactory('costID' , contentType=InputTypes.StringType))
+    inputSpecification.addSub(InputData.parameterInputFactory('valueID', contentType=InputTypes.StringType))
+    return inputSpecification
+
+  def initialize(self, runInfo, inputs, initDict) :
+    """
+      Method to initialize the PostProcessor
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+      @ Out, None
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
+
+  def _localReadMoreXML(self, xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.Element, Xml element node
+      @ Out, None
+    """
+    paramInput = ParetoFrontier.getInputSpecification()()
+    paramInput.parseNode(xmlNode)
+    self._handleInput(paramInput)
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the parsed paramInput for this class.
+      @ In, paramInput, ParameterInput, the already-parsed input.
+      @ Out, None
+    """
+    costID  = paramInput.findFirst('costID')
+    self.costID  = costID.value
+    valueID = paramInput.findFirst('valueID')
+    self.valueID = valueID.value
+
+  def inputToInternal(self, currentInp):
+    """
+      Method to convert an input object into the internal format that is
+      understandable by this pp.
+      In this case, we only want data objects!
+      @ In, currentInp, list, an object that needs to be converted
+      @ Out, currentInp, DataObject.HistorySet, input data
+    """
+    if len(currentInp) > 1:
+      self.raiseAnError(IOError, 'ParetoFrontier postprocessor {} expects one input DataObject, but received {} inputs!".'
+                                  .format(self.name,len(currentInp)))
+    currentInp = currentInp[0]
+    if currentInp.type not in ['PointSet']:
+      self.raiseAnError(IOError, 'ParetoFrontier postprocessor "{}" requires a DataObject input! Got "{}".'
+                                 .format(self.name, currentInput.type))
+    return currentInp
+
+  def run(self, inputIn):
+    """
+      This method executes the postprocessor action.
+      @ In, inputIn, DataObject, point set that contains the data to be processed
+      @ Out, paretoFrontierDict, dict, dictionary containning the Pareto Frontier information
+    """
+    inData = self.inputToInternal(inputIn)
+    data = inData.asDataset()
+    sortedData = data.sortby(self.costID)
+
+    coordinates = np.zeros(1,dtype=int)
+    for index,elem in enumerate(sortedData[self.costID].values):
+      if (index>1) and (sortedData[self.valueID].values[index]>sortedData[self.valueID].values[coordinates[-1]]):
```

Comment:
can't you use ```xarray.where``` and then get the coordinates directly? this loop can be super slow if we have a lot of data.

Reply:
The second condition in the IF statement compare an element of the array and its predecessor, not sure how to implement with WHERE

Reply:
Fixed!